sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -652,3 +652,15 @@ case object UnresolvedSeed extends LeafExpression with Unevaluable {
   override def dataType: DataType = throw new UnresolvedException("dataType")
   override lazy val resolved = false
 }
+
+/**
+ * An intermediate expression to hold a resolved (nested) column. Some rules may need to undo the
+ * column resolution and use this expression to keep the original column name.
+ */
+case class TempResolvedColumn(child: Expression, nameParts: Seq[String]) extends UnaryExpression
```

Comment:
Is its child always a named expression?

Reply:
Yes, I'll refine the type.

Reply:
Actually, we don't require it to be a named expression, so it's more robust to keep the type as `Expression`.